### PR TITLE
Add delete endpoint

### PR DESCRIPTION
**Pull request by Kristin01**
Add a delete function of the Promotions.
https://github.com/2020-nyu-devops/promotions/issues/4

### CODE REVIEW

**Comment by VickySteeves** [comment]
hey @Kristin01 , this PR has the code for the `update` function as well as the `delete` function. If you get rid of the `update` code in this (because it's duplicate, with your other PR #23 , and we should keep the two separate like you did :smile: ).

File: service/service.py [hunk 1]
```diff
@@ -88,6 +88,44 @@ def list_promotions():
...
+@app.route("/promotions/<int:promotion_id>", methods=["DELETE"])
+def delete_promotions(promotion_id):
+    """
+    Delete a Promotions
+    This endpoint will delete a Promotion based the id specified in the path
+    """
+    app.logger.info("Request to delete promotion with id: %s", promotion_id)
+    promotion = Promotion.find(promotion_id)
+    if not promotion:
+        return make_response("", status.HTTP_200_OK)
```
Comment (VickySteeves) [inline service/service.py:123]: This should be a 404 status

**Comment by VickySteeves** [comment]
Well I tested the `delete` function by first adding two promotions from `01-halloween.json` and `02-thanksgiving.json`. I went to `http://0.0.0.0:5000/promotions` to double check they were both there, and they were. 

So then I used CURL to DELETE the second promotion, with `id:2` (the thanksgiving promotion): 
```
curl -X DELETE http://127.0.0.1:5000/promotions/02
``` 

Nothing printed out after that, so I had no idea if it worked. I went back to `http://0.0.0.0:5000/promotions`  and indeed the second promotion was gone, leaving only the first one. You should probably change the status code in line #123 of the `services.py` file to 404, also.

Otherwise, the `delete` function works as expected! I ran `nosetests` and we had 92% coverage so that's all good too.

So once you delete the code for the `update` function that is in this PR and fix the status code in line 123 of the `services.py` file, I can merge it!

File: service/service.py (hunk 1 above)
Comment (Kristin01) [reply service/service.py:123]: Hi, Vicky, thank you for your commit. The acceptance criteria of the issue sets that if you call the DELETE function on a promotion that doesn't exist, should return OK, so I set this to 200_OK status. If it is better to set to 404 status, do we need to set a fix to the issue then change it, or could we just change it in the same issue?

Comment (VickySteeves) [reply service/service.py:123]: Ah I missed that, you're right! Nevermind :sweat_smile:

Comment (Kristin01) [reply service/service.py:123]: 😊

**Comment by Kristin01** [comment]
Hi Vicky, I just modify the code and request a new PR, you can merge it in the new PR now😊

**Pull request closed without merging**